### DISCUSSION TITLE
gRPC Protobuf definition failed to generate classes for Windows head but succeeds for Android. Can't compile

### DISCUSSION THREAD

**Discussion by boogagiga2** [unoplatform/uno#14859] [Q&A] [answered]
Using Visual Studio 2022 17.8.3, I've created a fresh Uno blank .NET 8 project with only the Theme enabled, and only Android and Windows platforms enabled. I separately set up a gRPC project from the built-in VS template, followed the MS docs on it which has you test the client connecting to the server running locally. Both server and client test projects were tested and there were no issues with the client project auto-generating classes that were then referenced within the manually written code. (had no issues at runtime as well)

Removed test client project, and followed the same instructions to add the protobuf info to the platform-independent Uno project so it could automatically generate the necessary classes for gRPC. The Android project (net8.0-android) created the necessary classes and compiled just fine, while Windows (net8.0-windows10.0.19041) was like 'Whoa, those generated classes don't exist!' and didn't compile

How do I resolve this? Thanks!

**Comment by limeniye** [comment]
Hello, petrushka.
Please, attach the sample with this issue.

**Reply by boogagiga2** [reply]
Hello limeniye, here's the sample. I've since also tried doing a git clean to force everything to rebuild from scratch but still getting the same issue.

[UnoTest.zip](https://github.com/unoplatform/uno/files/13693571/UnoTest.zip)

**Reply by limeniye** [reply, marked as answer]
It doesn't make sense to connect your .protos to UI project.
The UI project doesn't need to know anything about these files. They must be encapsulated at the level of business logic.
You should have created a separate **netstandard** project. In this case, your problem will be solved.
But if you fundamentally need to do exactly as you want, then you can use the following way:
```XML
    <Protobuf Include="Protos\transport.proto" GrpcServices="Client"
              OutputDir="Generated" CompileOutputs="true"/>
```

**Reply by boogagiga2** [reply]
Thanks for the advice, separating to a different project made it compile!

A heads up though, the alternate method of adjusting the Protobuf line ended up causing compilation errors due to duplicate definitions. No problem for me as I don't have to keep it that way.
